src/pm/hydra/proxy/proxy.c

```diff
@@ -0,0 +1,801 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2017 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#include "hydra.h"
+#include "mpx.h"
+#include "proxy.h"
+
+/* *INDENT-OFF* */
+struct proxy_params proxy_params = {
+    .debug = 0,
+    .tree_width = -1,
+
+    .root = {
+        .proxy_id = -1,
+        .node_id = -1,
+        .upstream_fd = -1,
+        .hostname = NULL,
+        .subtree_size = -1,
+        .barrier_ref_count = -1,
```

Comment:
Should this be initialized to `0` so we don't get stuck waiting in `proxy_barrier_in` for the final barrier participant?

We wait for `proxy_params.immediate.proxy.num_children + proxy_params.immediate.process.num_children` where the proxy itself doesn't participate. So if you have one local child and no child proxies, you'll wait for `ref_count` to be `1`, but it will only increment once, taking it to `0`.